Auditing Composition UI

satkunas opened the pull request:
# Description
Refactor Auditing to Vue3 composition.

# Delete branch after merge
YES 

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## New Features

## Enhancements
* Advanced search and CSV export in auditing

cgx commented on html/pfappserver/root/src/views/Auditing/_router/index.js:
```diff
@@ -1,179 +1,27 @@
+import DhcpOption82LogsRoutes from '../dhcpOption82Logs/_router'
+import DnsLogsRoutes from '../dnsLogs/_router'
+import RadiusLogsRoutes from '../radiusLogs/_router'
+import LiveLogsRoutes from '../liveLogs/_router'
```
This file is missing.

satkunas replied:
Missing file added in a4f46426f4a23dbca9ff28c0e109ee1bc541c435

cgx commented on html/pfappserver/root/src/views/Auditing/radiusLogs/_components/TheSearch.vue:
```diff
@@ -1,3 +1,153 @@
+}
+</script>
+
+<!--
```
Can you remove the commented-out code?

cgx commented on html/pfappserver/root/src/views/Auditing/radiusLogs/_components/TheView.vue:
```diff
@@ -0,0 +1,140 @@
+  <b-form @submit.prevent ref="rootRef">
+    <b-card no-body>
+      <b-card-header>
+        <b-button-close @click="close" v-b-tooltip.hover.left.d300 :title="$t('Close [ESC]')"><icon name="times"></icon></b-button-close>
```
`close` doesn't seem to be defined.

cgx commented:
Listing the search criteria is useless when listing *all* possible fields.

![placeholders-storm](https://user-images.githubusercontent.com/1088448/124531022-9b6b2a00-dddb-11eb-80b2-2667f7ee447b.jpg)

cgx commented:
The initial state of the advanced search is no longer matching the criteria of the simple search — I understand that it could be intense since we perform a search on *all* possible fields. However, the padding of the form input fields seems wrong.

![advanced-search-initial-state](https://user-images.githubusercontent.com/1088448/124531211-10d6fa80-dddc-11eb-9690-9f6b052195ea.jpg)

cgx commented:
When I look at a single DNS log entry, I get an *undefined* error. Could be related to my dataset.

![dns-audit-log-undefined-identifier](https://user-images.githubusercontent.com/1088448/124531443-85119e00-dddc-11eb-9392-9953e53798e1.jpg)

cgx commented:
The toggle button on the live logs page is misaligned with the label.

![live-logs-re-alignment](https://user-images.githubusercontent.com/1088448/124531532-b4280f80-dddc-11eb-8da2-50c4620a7ceb.jpg)

cgx commented:
The save searches that appear under the search button and under the navigation item don't match.

![save-search-2](https://user-images.githubusercontent.com/1088448/124531983-9c04c000-dddd-11eb-993a-1f22e3535d24.jpg)
![save-search-1](https://user-images.githubusercontent.com/1088448/124531985-9c9d5680-dddd-11eb-9e31-8262150b0e18.jpg)

satkunas commented:
> Listing the search criteria is useless when listing _all_ possible fields.
> 
> ![placeholders-storm](https://user-images.githubusercontent.com/1088448/124531022-9b6b2a00-dddb-11eb-80b2-2667f7ee447b.jpg)

I've reworked the placeholder to show a maximum of 3 item, while also adding a title (w/ hover) that expands all the criteria. 

![Screen Shot 2021-07-06 at 10 38 06 AM](https://user-images.githubusercontent.com/3904468/124629769-bd9e8f80-de4f-11eb-816e-2aa616c8192f.jpg)

satkunas commented:
> The toggle button on the live logs page is misaligned with the label.
> 
> ![live-logs-re-alignment](https://user-images.githubusercontent.com/1088448/124531532-b4280f80-dddc-11eb-8da2-50c4620a7ceb.jpg)

Fixed in 0b008f8c4792ff94d87c24261f8c5e9c0a2819ea

satkunas commented:
> The initial state of the advanced search is no longer matching the criteria of the simple search — I understand that it could be intense since we perform a search on _all_ possible fields. However, the padding of the form input fields seems wrong.
> 
> ![advanced-search-initial-state](https://user-images.githubusercontent.com/1088448/124531211-10d6fa80-dddc-11eb-9690-9f6b052195ea.jpg)

I don't like expanding all the available criteria as the default search condition. Instead I added a placeholder on a single field using the ID (or MAC).

satkunas commented:
> When I look at a single DNS log entry, I get an _undefined_ error. Could be related to my dataset.
> 
> ![dns-audit-log-undefined-identifier](https://user-images.githubusercontent.com/1088448/124531443-85119e00-dddc-11eb-9392-9953e53798e1.jpg)

I'm not able to reproduce. This appears to be your dataset.

satkunas commented:
> The save searches that appear under the search button and under the navigation item don't match.
> 
> ![save-search-2](https://user-images.githubusercontent.com/1088448/124531983-9c04c000-dddd-11eb-993a-1f22e3535d24.jpg)
> ![save-search-1](https://user-images.githubusercontent.com/1088448/124531985-9c9d5680-dddd-11eb-9e31-8262150b0e18.jpg)



This will be resolved later. I still need to unify the 3 user-preference store modules into a single module (either VueX or Pinia), as well as Vue3-ize all the index.vue for the sidebar content. I will resolve both at the same time. I will also try to find a more elegant way to display the saved searches in the sidebar without using as much vertical space. One idea is to include a save-search icon in the sidebar item that doubles as a dropdown containing all the save searches. Thoughts?

satkunas commented:
@cgx all comments addressed. Please follow-up.

cgx commented:
> I will also try to find a more elegant way to display the saved searches in the sidebar without using as much vertical space. One idea is to include a save-search icon in the sidebar item that doubles as a dropdown containing all the save searches. Thoughts?

I think it is convenient to have the saved searches accessible from the sidebar. Having them visible without having to expand a list also saves a mouse click. To save space, we could remove the *Saved Searches* header and put a small magnifier icon on the left side of each saved search.

cgx commented:
> I've reworked the placeholder to show a maximum of 3 item, while also adding a title (w/ hover) that expands all the criteria.
> 
> ![Screen Shot 2021-07-06 at 10 38 06 AM](https://user-images.githubusercontent.com/3904468/124629769-bd9e8f80-de4f-11eb-816e-2aa616c8192f.jpg)

I wonder if this is just too much information? Since the search will cover all fields, the results should be no surprise for the user. The placeholder could simply be *Search criteria*.

cgx commented:
> I don't like expanding all the available criteria as the default search condition. Instead I added a placeholder on a single field using the ID (or MAC).

The padding is still incorrect (missing) between the input fields.

satkunas commented:
> > I've reworked the placeholder to show a maximum of 3 item, while also adding a title (w/ hover) that expands all the criteria.
> > ![Screen Shot 2021-07-06 at 10 38 06 AM](https://user-images.githubusercontent.com/3904468/124629769-bd9e8f80-de4f-11eb-816e-2aa616c8192f.jpg)
> 
> I wonder if this is just too much information? Since the search will cover all fields, the results should be no surprise for the user. The placeholder could simply be _Search criteria_.

I've reworked the placeholder and tooltip as a compromise.

![Screen Shot 2021-07-07 at 1 30 58 PM](https://user-images.githubusercontent.com/3904468/124803906-ae8b1080-df27-11eb-8e22-85fb894dc389.jpg)
![Screen Shot 2021-07-07 at 1 31 26 PM](https://user-images.githubusercontent.com/3904468/124803917-b1860100-df27-11eb-81c2-d5512c1b2499.jpg)
